airbyte-lib/tests/integration_tests/test_snowflake_cache.py

```diff
@@ -0,0 +1,148 @@
+from pathlib import Path
+
+import pytest
+import ulid
+import viztracer
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+
+import airbyte_lib as ab
+from airbyte_lib import caches
+
+
+# Product count is always the same, regardless of faker scale.
+NUM_PRODUCTS = 100
+
+SEED_A = 1234
+SEED_B = 5678
+
+# Number of records in each of the 'users' and 'purchases' streams.
+FAKER_SCALE_A = 200
+# We want this to be different from FAKER_SCALE_A.
+FAKER_SCALE_B = 300
+
+
+# Patch PATH to include the source-faker executable.
+
+@pytest.fixture(autouse=True)
+def add_venv_bin_to_path(monkeypatch):
+    # Get the path to the bin directory of the virtual environment
+    venv_bin_path = os.path.join(sys.prefix, 'bin')
+
+    # Add the bin directory to the PATH
+    new_path = f"{venv_bin_path}:{os.environ['PATH']}"
+    monkeypatch.setenv('PATH', new_path)
+
+
+@pytest.fixture(scope="function")  # Each test gets a fresh source-faker instance.
+def source_faker_seed_a() -> ab.Source:
+    """Fixture to return a source-faker connector instance."""
+    source = ab.get_source(
+        "source-faker",
+        local_executable="source-faker",
+        config={
+            "count": FAKER_SCALE_A,
+            "seed": SEED_A,
+            "parallelism": 16,  # Otherwise defaults to 4.
+        },
+        install_if_missing=False,  # Should already be on PATH
+    )
+    source.check()
+    source.select_streams([
+        "users",
+    ])
+    return source
+
+
+@pytest.fixture(scope="function")  # Each test gets a fresh source-faker instance.
+def source_faker_seed_b() -> ab.Source:
+    """Fixture to return a source-faker connector instance."""
+    source = ab.get_source(
+        "source-faker",
+        local_executable="source-faker",
+        config={
+            "count": FAKER_SCALE_B,
+            "seed": SEED_B,
+            "parallelism": 16,  # Otherwise defaults to 4.
+        },
+        install_if_missing=False,  # Should already be on PATH
+    )
+    source.check()
+    source.select_streams([
+        "users",
+    ])
+    return source
+
+
+@pytest.fixture(scope="function")
+def snowflake_cache(snowflake_config) -> Generator[caches.SnowflakeCache, None, None]:
+    """Fixture to return a fresh cache."""
+    cache: caches.SnowflakeCache = caches.SnowflakeSQLCache(snowflake_config)
+    yield cache
+    # TODO: Delete cache DB file after test is complete.
+    return
+
+
+@viztracer.trace_and_save(output_dir=".pytest_cache/snowflake_trace/")
+def test_faker_read_to_snowflake(
+    source_faker_seed_a: ab.Source,
+    snowflake_cache: ab.SnowflakeCache,
+) -> None:
+    """Test that the append strategy works as expected."""
+    result = source_faker_seed_a.read(
+        snowflake_cache, write_strategy="replace", force_full_refresh=True
+    )
+    assert len(list(result.cache.streams["users"])) == FAKER_SCALE_A
+
+
+def test_replace_strategy(
+    source_faker_seed_a: ab.Source,
+    snowflake_cache: ab.DuckDBCache,
```

Comment:
nit. should the parameter name be duckDB_cache ?

Reply:
Good catch! Thanks!

```suggestion
    snowflake_cache: ab.SnowflakeCache,
```